src/aznfswatchdog

```diff
@@ -0,0 +1,190 @@
+
+    #
+    # TODO: Add a function reconcile_mountmap() and call it from here. This
+    #       should reconstruct the MOUNTMAP file from findmnt and output of
+    #       iptables. This will be added in subsequent revisions.
+    #
+
+    findmnt=$(findmnt --raw --noheading -o MAJ:MIN,FSTYPE,SOURCE,TARGET,OPTIONS -t nfs 2>&1)
+
+    #
+    # For no matching mounts also, findmnt exits with a failure return, so check
+    # for both exit status and non-empty error o/p.
+    #
+    if [ $? -ne 0 -a -n "$findmnt" ]; then
+        eecho "findmnt failed!"
+        eecho "${findmnt}."
+        # This usually indicates some non-transient issue, bail out.
+        exit 1
+    fi
+
+    #
+    # Go over all lines in $MOUNTMAP and check them for two things:
+    # 1. Is that entry still mounted, if not remove the entry.
+    # 2. Has the Blob endpoint address changed from what is stored?
+    #    If yes, update DNAT rule to point to the new address.
+    #
+    # Sample line in $MOUNTMAP.
+    # account.blob.preprod.core.windows.net:/testaccount/testcontainer 10.100.100.100 52.230.170.200
+    #
+    # where the format is
+    # <blobendpoint>:/account/container LOCAL_IP blobendpoint_ip
+    # 
+    IFS=$'\n' lines=$(cat $MOUNTMAP) 
+
+    for line in $lines; do
+        if [ -z "$line" ]; then
+            continue
+        fi
+
+        l_share=$(echo "$line" | awk '{print $1}')
+        l_host=$(echo "$l_share" | cut -d: -f1)
+        l_dir=$(echo "$l_share" | cut -d: -f2)
+        l_ip=$(echo "$line" | awk '{print $2}')
+        l_nfsip=$(echo "$line" | awk '{print $3}')
+        l_pid=$(echo "$line" | awk '{print $4}')
+
+        if [ -z "$l_host" -o -z "$l_dir" -o -z "$l_ip" -o -z "$l_nfsip" ]; then
+            wecho "[FATAL] Deleting invalid line in $MOUNTMAP: [$line]."
+            if ! ensure_mountmap_not_exist "$line"; then
+                eecho "Failed to delete [$line] from MOUNTMAP."
+            fi
+            continue
+        fi
+
+        # Since we added it to the MOUNTMAP file, it cannot be invalid.
+        if ! is_private_ip "$l_ip"; then
+            wecho "[FATAL] local ip ($l_ip) is invalid!"
+            if ! ensure_mountmap_not_exist "$line"; then
+                eecho "Failed to delete [$line] from MOUNTMAP."
+            fi
+            continue
+        fi
+
+        # Since we added it to the MOUNTMAP file, it cannot be invalid.
+        if ! is_valid_ipv4_address "$l_nfsip" || is_private_ip "$l_nfsip"; then
+            wecho "[FATAL] Blob endpoint ip ($l_nfsip) is invalid!"
+            if ! ensure_mountmap_not_exist "$line"; then
+                eecho "Failed to delete [$line] from MOUNTMAP."
+            fi
+            continue
+        fi
+
+        #
+        # This entry was added by get_free_local_ip() call in mount.aznfs to
+        # to ensure two parallel mount requests do not get same local ip. Skip
+        # this entry if PID is active.
+        #
+        if [ ! -z $l_pid ]; then
```

Comment:
if [ -n "$l_pid" ]; then

use of quotes is very important, else shell will give error